### PR TITLE
DMIC: Fixed length IPC

### PR DESCRIPTION
**Pull request by singalsu**

### CODE REVIEW

**Comment by singalsu** [comment]
There's an open with ABI minor version. Seems like firmware is ahead by one in minor version and I think the FW and kernel side PRs for DMIC IPC change should propose the same ABI minor version.

**Comment by singalsu** [comment]
There's good feedback for kernel side PR. I'll soon update both PRs the same way.

**Comment by singalsu** [comment]
New version to sync with kernel. I changed the macro name to SOF_DAI_INTEL_DMIC_NUM_CTRL. It helps to avoid to confuse this with microphones or channels count.

File: src/include/ipc/dai-intel.h [hunk 1]
```diff
@@ -192,8 +196,8 @@ struct sof_ipc_dai_dmic_params {
 	/* reserved for future use */
 	uint32_t reserved[5];
 
-	/**< variable number of pdm controller config */
-	struct sof_ipc_dai_dmic_pdm_ctrl pdm[0];
+	/**< Fixed number of pdm controller config */
+	struct sof_ipc_dai_dmic_pdm_ctrl pdm[SOF_DAI_INTEL_DMIC_NUM_CTRL];
```
Comment (lgirdwood) [inline src/include/ipc/dai-intel.h]: Whats the rational behind this change, it's missing from commit message.

Comment (singalsu) [reply src/include/ipc/dai-intel.h]: Yep, need to add.

There's fair amount of comments on kernel side and I'll do the same updates for firmware too.

File: src/include/kernel/abi.h [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 /** \brief SOF ABI version major, minor and patch numbers */
 #define SOF_ABI_MAJOR 3
-#define SOF_ABI_MINOR 14
+#define SOF_ABI_MINOR 15
```
Comment (lgirdwood) [inline src/include/kernel/abi.h]: lets say 14, others pending on 14 too.

Comment (singalsu) [reply src/include/kernel/abi.h]: OK, I'll remove ABI minor increase from next version. I'll still wait for new review round comments for kernel side to see if the changes were OK before I update this.

Comment (singalsu) [reply src/include/kernel/abi.h]: Done

File: src/include/ipc/dai-intel.h (hunk 1 above)
Comment (singalsu) [reply src/include/ipc/dai-intel.h]: Commit text is now updated and header file comments are in sync with kernel.

Comment (singalsu) [reply src/include/ipc/dai-intel.h]: Sorry, diffing revealed a difference of one "." so new version again. There's also lot of other changes but I'll complete my part.

Comment (singalsu) [reply src/include/ipc/dai-intel.h]: Done. Finally same DMIC configuration headers as in kernel PR version.

**Comment by lgirdwood** [comment]
SOFCI TEST

File: src/include/ipc/dai-intel.h [hunk 3]
```diff
@@ -146,7 +146,7 @@ struct sof_ipc_dai_dmic_pdm_ctrl {
  * version number used in configuration data is checked vs. version used by
  * device driver src/drivers/dmic.c need to match. It is incremented from
  * initial value 1 if updates done for the to driver would alter the operation
- * of the microhone.
```
Comment (ranj063) [inline src/include/ipc/dai-intel.h]: This I found hilarious :) you left a typo in the word typo itself in the commit message

**Comment by lgirdwood** [comment]
Jenkins all green

File: src/include/ipc/dai-intel.h (hunk 3 above)
Comment (singalsu) [reply src/include/ipc/dai-intel.h]: Haha, I only now noticed it!! Can I still fix this without restarting the approval process?

Comment (singalsu) [reply src/include/ipc/dai-intel.h]: @lgirdwood @kv2019i Should I fix this?

Comment (ranj063) [reply src/include/ipc/dai-intel.h]: @singalsu no need I think. its just the commit message.

Comment (singalsu) [reply src/include/ipc/dai-intel.h]: Good, I'll leave this funny bit into the history.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@singalsu if both this PR and kernel one are approved we can merge at 15 ? Providing bot PRs dont change the ABI minor (which is already at 15). @kv2019i kernel PR good for you at 15 too ?

**Comment by kv2019i** [comment]
@lgirdwood wrote:
> @singalsu if both this PR and kernel one are approved we can merge at 15 ? Providing bot PRs dont change the ABI minor (which is already at 15). @kv2019i kernel PR good for you at 15 too ?

Ack, the kernel side PR (https://github.com/thesofproject/linux/pull/1924) is all good to go. Just needs to be refreshed on latest sof-dev.

**Comment by kv2019i** [comment]
@lgirdwood Are we good to go? Kernel side is approved and ready https://github.com/thesofproject/linux/pull/1924

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@kv2019i sorry, it looks like this one was missing from full CI tests. rerunning.

**Comment by lgirdwood** [comment]
Travis all green, but showing pending. Jenkins known issues. @kv2019i merged, good for kernel side now.

**Comment by kv2019i** [comment]
> Travis all green, but showing pending. Jenkins known issues. @kv2019i merged, good for kernel side now.

Thanks @lgirdwood , kernel merged now as well.

**Comment by lgirdwood** [comment]
@kv2019i great :) I will tag v1.5-rc1 now.

**Pull request merged**